src/performance.js

```diff
@@ -60,6 +62,44 @@ export class Performance {
     this.events_ = [];
   }
 
+  /**
+   * Listens to viewer and resource events.
+   */
+  coreServicesAvailable() {
+    const viewer = viewerFor(this.win);
+    const resources = resourcesFor(this.win);
+
+    viewer.onVisibilityChanged(this.flush.bind(this));
+
+    // -1 if still in prerender mode
+    let visibleTime = viewer.hasBeenVisible() ? timer.now() : -1;
+    let visualCompletenessTime = -1;
+
+    // Only listen to visibility request if doc is still prerendering, since
+    // if it isnt prerendering, User probably already requested for it
+    // to be visible.
+    if (visibleTime == -1) {
+      // User requested document to be visible.
+      viewer.whenFirstVisible().then(() => {
+        visibleTime = timer.now();
+      });
+    }
+
+    // First viewport done.
+    resources.whenFirstViewportLoaded().then(() => {
+      // Only set visualCompleteness to the current time if we have the time
+      // when the User requested document to be visible. This is so that we can
+      // tick `0` if if we've loaded the viewport before User requested
+      // doc to be visible.
+      if (visibleTime != -1) {
+        visualCompletenessTime = timer.now();
```

Comment:
I don't think so, but might misunderstand.

The third argument to tick must be a delta, right? When you don't provide it, the delta since page load start until the tick is calculated.